From 4936ca25f3e9a2875ce063033e5e8f3a3590dab2 Mon Sep 17 00:00:00 2001
From: Davide Madrisan <davide.madrisan@gmail.com>
Date: Mon, 27 Jan 2020 22:00:41 +0100
Subject: [PATCH] The build fails when '-fno-common' is added to CFLAGS

As reported by 'sbraz', the build stops with the error message:

   (.bss+0x8): multiple definition of `program_name'
   (.bss+0x0): multiple definition of `program_name_short'

This flag will be apparently enabled by default in gcc 10.

Fix this build problem by correctly referencing as extern both
the variables in the header file.

Signed-off-by: Davide Madrisan <davide.madrisan@gmail.com>
---
 include/progname.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/progname.h b/include/progname.h
index 576ed16..847122a 100644
--- a/include/progname.h
+++ b/include/progname.h
@@ -25,10 +25,10 @@ extern "C" {
 #endif
 
   /* String containing name the program is called with.  */
-  const char *program_name;
+  extern const char *program_name;
 
   /* String containing a short version of 'program_name'.  */
-  const char *program_name_short;
+  extern const char *program_name_short;
 
   /* Set program_name, based on argv[0].
      argv0 must be a string allocated with indefinite extent, and must not be
